features/environment.py

```diff
@@ -98,36 +98,57 @@ def from_environ(cls) -> "UAClientBehaveConfig":
 
 
 def before_all(context: Context) -> None:
-    """behave will invoke this before anything else happens.
-
-    In this function, we launch a container, install ubuntu-advantage-tools and
-    then capture an image.  This image is then reused by each feature, reducing
-    test execution time.
-    """
+    """behave will invoke this before anything else happens."""
     userdata = context.config.userdata
     context.reuse_container = userdata.get("reuse_container")
+    context.series = {}
+    context.image_name = ""
     context.config = UAClientBehaveConfig.from_environ()
-    if context.config.reuse_image is None:
-        create_trusty_uat_lxd_image(context)
-    else:
-        context.image_name = context.config.reuse_image
 
 
-def before_scenario(context: Context, scenario: Scenario):
-    for tag in scenario.effective_tags:
+def before_feature(context: Context, feature: Feature):
+    for tag in feature.tags:
         parts = tag.split(".")
         if parts[0] == "uses":
             val = context
             for attr in parts[1:]:
                 val = getattr(val, attr, None)
                 if val is None:
-                    scenario.skip(
+                    feature.skip(
                         reason="Skipped because tag value was None: {}".format(
                             tag
                         )
                     )
 
 
+def before_scenario(context: Context, scenario: Scenario):
+    """
+    In this function, we launch a container, install ubuntu-advantage-tools and
+    then capture an image. This image is then reused by each scenario, reducing
+    test execution time.
+    """
+    for tag in scenario.effective_tags:
+        parts = tag.split(".")
+        if parts[0] == "series":
+            series = parts[1]
+            if context.config.reuse_image is None:
```

Comment:
There is a problem now with reuse_image because any image you provide will be used for both trusty and focal. I'm not sure how best to solve this. Maybe if reuse_image is provided you have to specify only one series?

Reply:
I've solved that. It will check the image series and will show it.
For instance, if you provide a trusty image and then run all tests (trusty and focal) it will use the trusty image for trusty tests and it will create a new one for focal.